api/v1alpha1/selfnoderemediation_types.go

```diff
@@ -46,7 +46,8 @@ type SelfNodeRemediationSpec struct {
 	// Automatic will choose the most appropriate strategy during runtime.
 	// +kubebuilder:default:="Automatic"
 	// +kubebuilder:validation:Enum=Automatic;ResourceDeletion;OutOfServiceTaint
-	RemediationStrategy RemediationStrategyType `json:"remediationStrategy,omitempty"`
+	// +required
```

Comment:
why is this needed, we have the default value? :thinking:

Reply:
I think there is value in making this field explicitly mandatory.

Even though as you've suggested it technically will not  be empty because of the default value, making it mandatory clears our intention as what we expect of this field.
This may also become relevant in case in the future we decide we want to remove the default value.

Do you see a specific downside for this that I'm missing  ?

Reply:
The obvious downside is: this is different than it's done at all other places with default values 🤷🏼‍♂️ 

The less obvious but significant downside is (I wasn't sure about it, but I just tried it in test code at least): the validation for mandatory fields runs before setting default values. So you can't create SNRs anymore without setting the strategy, which makes the default value useless. Try it by removing the strategy in modified `client_server_test.go`.